Neos.ContentRepository/Classes/Domain/Model/Node.php

```diff
@@ -960,11 +960,15 @@ public function getProperty($propertyName, bool $returnNodesAsIdentifiers = fals
      */
     protected function resolvePropertyReferences(array $value = []): array
     {
-        $nodes = array_map(function ($nodeIdentifier) {
-            return $this->context->getNodeByIdentifier($nodeIdentifier);
-        }, $value);
-
-        return array_filter($nodes);
+        $nodes = [];
+        foreach ($value as $nodeIdentifier) {
+            $node = $this->context->getNodeByIdentifier($nodeIdentifier);
+            // $node can be NULL if the node is not visible (or removed) according to the current content context:
+            if ($node !== null) {
+                $nodes[] = $node;
+            }
+        }
+        return $nodes;
```

Comment:
Just adding an `array_values()` around the return would work, as well:
```suggestion
        $nodes = array_map(function ($nodeIdentifier) {
            return $this->context->getNodeByIdentifier($nodeIdentifier);
        }, $value);

        return array_values(array_filter($nodes));
```

Reply:
yes thanks i know ;) But found it more funny to reuse the 8 year old original pattern - which also seems more intuitive ^^